test/stakePool.test.ts

```diff
@@ -0,0 +1,90 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { loadFixture, time } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { DAY_SECONDS } from './helpers/constants';
+import { deployStakedBnbTokenFixture, depositBnbFixture, setupProtocolFixture } from './helpers/fixtures';
+import { AllContractsDeployment } from './helpers/fixtures/types';
+
+describe("Stakepool tests", function () {
+  it("should pause the StakedBNBToken", async () => {
+    const stakedBNBTokenDeployment = await loadFixture(deployStakedBnbTokenFixture);
+    const { stakedBNBToken } = stakedBNBTokenDeployment;
+    expect(await stakedBNBToken.paused()).to.be.false;
+    expect(await stakedBNBToken.pause()).to.not.be.reverted;
+    expect(await stakedBNBToken.paused()).to.be.true;
+  });
+
+  describe("deposit tests", () => {
+    it("should make a deposit", async () => {
+      const allContractsDeployment = await loadFixture(setupProtocolFixture);
+      const { stakePool } = allContractsDeployment.contracts;
+      const [user] = await ethers.getSigners();
+      const stakePoolConnectedUser = stakePool.connect(user);
+      expect(await stakePoolConnectedUser.deposit({ value: ethers.constants.WeiPerEther })).to.not.be.reverted;
+    });
+  
+    it("should mint stkBNB correctly", async () => {
```

Comment:
Can you add slashing and reward distribution scenario as well. When the rewards are claimed on bnb side and updated on contracts, the price for stkBNB changes and new users get lesser stkBNB